Add `id-length` ESLint rule

ehmicky opened the pull request:
This adds the [`id-length`](https://eslint.org/docs/rules/id-length) ESLint rule.
See also https://github.com/netlify/build/pull/1982

Single-letter variable names makes reading code less clear.

erezrokah reviewed (changes requested):
This is huge @ehmicky thanks.
Added a few comments

ehmicky commented:
Answered this in the PR for Netlify Build, let's solve this first: https://github.com/netlify/build/pull/1982#issuecomment-713112437

ehmicky commented on src/commands/dev/index.js:
```diff
@@ -92,7 +92,7 @@ async function startFrameworkServer({ settings, log, exit }) {
   return ps
 }
 
-const getAddonsUrlsAndAddEnvVariablesToProcessEnv = async ({ api, site, flags }) => {
```
:+1:

ehmicky commented on src/functions-templates/js/identity-signup/identity-signup.js:
```diff
@@ -16,7 +16,7 @@ exports.handler = async function (event) {
     user_metadata: {
       ...user.user_metadata, // append current user metadata
-      custom_data_from_function: 'hurray this is some extra metadata',
+      custom_data_from_func: 'hurray this is some extra metadata',
```
Good point :+1:

ehmicky commented:
Thanks @erezrokah! 
I have removed the `max` constraint.
I also removed all the changes in this PR related to long variable names, so this PR is a little smaller.

ehmicky commented:
Rebased. Does this look good @erezrokah ?
